### ISSUE TITLE
_clock_gettime Symbol not found on Mac 10.11

### ISSUE THREAD

**Issue by MikeMitterer** [pocoproject/poco#1453]
I just installed poco via brew on my mac.

``` bash
 brew install --c++11 poco
```

After running a little sample application if gives me

```
dyld: lazy symbol binding failed: Symbol not found: _clock_gettime
  Referenced from: /usr/local/opt/poco/lib/libPocoFoundation.45.dylib
  Expected in: /usr/lib/libSystem.B.dylib
```

OS ---------------------------------------------------------------------------
ProductName:    Mac OS X
ProductVersion: 10.11.6
BuildVersion:   15G31
Darwin Kernel Version 15.6.0: Thu Jun 23 18:25:34 PDT 2016; root:xnu-3248.60.10~1/RELEASE_X86_64

**Comment by obiltschnig** [comment]
Try building from source.

**Comment by obiltschnig** [comment]
For me, a little sample using the brew-installed POCO works fine. clock_gettime() is not available on OS X, so it's a bit strange.

Anyway, as first suggested, try building POCO yourself from the official source packages.

**Comment by bitjeep** [comment]
clock_gettime() was added in OS X 10.12.  You need to build against an earlier SDK if you want to be backwards compatible.

**Comment by obiltschnig** [comment]
Similar issue with iOS 10 SDK and iOS 9: https://pocoproject.org/forum/viewtopic.php?f=12&t=6878&p=12801#p12801

**Comment by gelldur** [comment]
Similar issue: http://stackoverflow.com/questions/39972455/libcurl-crashes-on-ios-9-when-compiled-with-latest-ios-sdk

**Comment by obiltschnig** [comment]
It seems this it's an issue with Apple's system headers. They indicate clock_gettime() is available with the MacOS 10.12 and iOS 10 SDKs even when specifying MacOS 10.11/iOS 9 or earlier as target (IPHONE_SDK_VERSION_MIN=9.0).

**Comment by bitjeep** [comment]
The minimum SDK version setting alone is not enough because Xcode is still building and linking against the current SDK (thus it can find the definition).  You need to use a NULL check for the symbol (function) at runtime:

https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/cross_development/Using/using.html
https://developer.apple.com/library/content/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/WeakLinking.html
